### ISSUE TITLE
Split style-spec to a different repo

### ISSUE THREAD

**Issue by HarelM** [maplibre/maplibre-gl-js#2149]
## Motivation
The style spec is a whole subject that deserves its own repo.
This new repo should facilitate for documentation and the style validation code.
It should also include the typescript definitions in that package (in npm).
This way, this repo can consume this package no matter how it is implemented internally.
As a starting point I would advise to simply move parts of this repo as is (without any code changes) and make sure everything works as expected, once this is done we can discuss the next step for this new repo, either creating a similar package for the native repo to use, refresh technology, change to json-scheme or whatever.
But the first step is to split this apart and define a proper interface.

## Design Alternatives

Keep everything together in a single repo.

## Design

Describe above - split to a different package, define an interface/API and use it

## Acceptance criteria for the bounty:
1. Style spec definition, validation, documentation are in a new repo, including CI of tests and publishing of package and docs.
2. The code is removed from this repo, and this repo only depends on packages produced by the other repo

Link to bounty:
https://github.com/maplibre/maplibre/issues/191

**Comment by HarelM** [comment]
Cc: @birkskyum @wipfli @louwers @ovivoda

**Comment by birkskyum** [comment]
These four issues contain quite a lot of gotchas and pitfalls. I'd be able to plan and resolve them all, but it would likely take me a full week, which I unfortunately can't carve out for a bounty this size ~25$/h. Let's see if someone faster wants to give it a go.

* (blocking for issues below) https://github.com/maplibre/maplibre-gl-js-docs/issues/384 
* (blocking for issues below) https://github.com/maplibre/maplibre-gl-js-docs/issues/389 
* https://github.com/maplibre/maplibre-gl-js/issues/2150
* https://github.com/maplibre/maplibre-gl-js/issues/2149

**Comment by HarelM** [comment]
Thanks for the summary of this. The current state is worse than I thought. I'll see what can be done.

**Comment by krishnaglodha** [comment]
Hi @HarelM , please let me know if the task is available, I'd like to contribute to it

**Comment by HarelM** [comment]
Please coordinate with @birkskyum as he has started looking into it and started working on it.
This issue is dependant on different issues and it's probably the last one to be finished.

**Comment by birkskyum** [comment]
@krishnaglodha , ping me in slack and i can give you a status on where we are

**Comment by birkskyum** [comment]
Style spec is split out from maplibre-gl-js repo already, so i consider this issue complete. It's unfair to turn this into an umbrella ticket after the fact just to withhold the bounty.

**Comment by birkskyum** [comment]
At least i don't hope anyone else are treated like that.

**Comment by HarelM** [comment]
It's not about an umbrella ticket, it's about acceptance criteria. The acceptance criteria is fairly clear from my point of view - spec repo has everything it needs to run "on its own". This is still not the case (as the docs are broken due to issues that were introduced). 
Once the [latest PR](https://github.com/maplibre/maplibre-gl-style-spec/pull/88) is reviewed and merged this issue can be closed, but not before that.

**Comment by birkskyum** [comment]
Okay. We could just remove that acceptance criteria though, since it's unrelated to this repo, and there's another ticket specifically for that. It's best to keep things separate.

**Comment by birkskyum** [comment]
Anyways if we can land the pr this should hopefully soon be done - I'm quite drained at this point as you can probably tell

**Comment by HarelM** [comment]
I'll try my best to review it soon so we can continue moving things forward. I'm sorry you feel this way and I think you did a great job. It's sometimes frustrating as you feel you invested a lot of energy and all you get is bad criticism, I'm sorry, and I'm probably the one to blame. it's not an excuse, but it's probably because I care a lot and want to do it as best as we can. Keep up the good work!!

**Comment by birkskyum** [comment]
Thanks, I'll try to push through then. I'm sure it'll be great - at least any alternative solution/stack I can imagine would be easier to achieve with this PR as the foundation.

**Comment by birkskyum** [comment]
I Guess this can be closed now then - unless the expression test conundrum also is included here.

**Comment by HarelM** [comment]
Resolved by https://github.com/maplibre/maplibre-style/pull/88

**Comment by wipfli** [comment]
@birkskyum let's add a link to the bounty payment in opencollective here. So you could write something like:

"Expense on open collective: https://opencollective.com/maplibre/expenses/132832"

And then @HarelM can comment something like "The work has been completed and the bounty can be paid out"

**Comment by wipfli** [comment]
https://maplibre.org/roadmap/step-by-step-bounties-guide/#3-claiming-a-bounty instructions are here

The doubly-linking might look like too much work, but I have seen a lot of expense requests on opencollective in other projects from people that did not do any work. So let us avoid scams and false payments by doing this doubly linking...
